diff --git a/node_modules/react-native-fetch-api/src/Fetch.js b/node_modules/react-native-fetch-api/src/Fetch.js
index 6879bcf..791ecaa 100644
--- a/node_modules/react-native-fetch-api/src/Fetch.js
+++ b/node_modules/react-native-fetch-api/src/Fetch.js
@@ -31,7 +31,7 @@ function createStream(cancel) {
 
 class Fetch {
     _nativeNetworkSubscriptions = new Set();
-    _nativeResponseType = "blob";
+    _nativeResponseType = "base64";
     _nativeRequestHeaders = {};
     _nativeResponseHeaders = {};
     _nativeRequestTimeout = 0;
